engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManager.scala

```diff
@@ -50,11 +53,21 @@ class K8sDeploymentManagerProvider extends DeploymentManagerProvider {
 
   override def createQueryableClient(config: Config): Option[QueryableClient] = None
 
-  override def typeSpecificInitialData: TypeSpecificInitialData = TypeSpecificInitialData(LiteStreamMetaData(Some(1)))
+  private val steamingInitialMetData = TypeSpecificInitialData(LiteStreamMetaData(Some(1)))
+
+  override def typeSpecificInitialData(config: Config): TypeSpecificInitialData = {
+    // TODO: mode field won't be needed if we add scenarioType to TypeSpecificInitialData.forScenario
+    //       and add scenarioType -> mode mapping with reasonable defaults to configuration
+    config.getString("mode") match {
```

Comment:
in case of no `mode` config at all, it will throw exception. but maybe we can live with that?

Reply:
it is "by design". To avoid situation when this exception will be thrown during deploying ...